pyproject.toml

```diff
@@ -35,7 +35,7 @@ classifiers = [
 ]
 
 [tool.poetry.dependencies]
-python = "^3.6"
+python = "^3.6.1"
```

Comment:
Looking at https://python-poetry.org/docs/cli/#options_3 - is `poetry add python=3.6.1 --dev --lock` what we're looking for?

Reply:
I tried a few different ways of expressing that (via the command, by adding it manually to `pyproject.toml`, etc.), none of which worked. They all result in something like 
```
Installing dependencies from lock file
Warning: The lock file is not up to date with the latest changes in pyproject.toml. You may be getting outdated dependencies. Run update to update them.

  SolverProblemError

  Because ward depends on python (^3.6.1) which doesn't match any versions, version solving failed.

  at ~/.poetry/lib/poetry/puzzle/solver.py:241 in _solve
      237│             packages = result.packages
      238│         except OverrideNeeded as e:
      239│             return self.solve_in_compatibility_mode(e.overrides, use_latest=use_latest)
      240│         except SolveFailure as e:
    → 241│             raise SolverProblemError(e)
      242│
      243│         results = dict(
      244│             depth_first_search(
      245│                 PackageNode(self._package, packages), aggregate_package_nodes
```
It seems like the `python` key is treated specially only in `tool.poetry.dependencies`. I poked around in poetry's internals a little but couldn't find where this happens. Will poke some more when I get a chance...

Reply:
I'm not opposed to just bumping the version of Python required, it's only a tiny patch bump and looking at e.g. Black - they only support 3.6.2+.